src/cascadia/TerminalConnection/ConptyConnection.cpp

```diff
@@ -657,15 +636,19 @@ namespace winrt::Microsoft::Terminal::TerminalConnection::implementation
 
             if (readFail) // reading failed (we must check this first, because read will also be 0.)
             {
+                // EXIT POINT
                 const auto lastError = GetLastError();
-                if (lastError != ERROR_BROKEN_PIPE)
+                if (lastError == ERROR_BROKEN_PIPE)
+                {
+                    _ClientTerminated();
```

Comment:
it might be best for us to rename `_ClientTerminated` to `_LastConPtyClientDisconnected` or something